core/src/avm2/globals/flash/display/displayobject.rs

```diff
@@ -682,6 +683,83 @@ pub fn set_blend_mode<'gc>(
     Ok(Value::Undefined)
 }
 
+fn scroll_rect<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(dobj) = this.and_then(|this| this.as_display_object()) {
+        if let Some(scroll_rect) = dobj.scroll_rect() {
+            return Ok(activation
+                .avm2()
+                .classes()
+                .rectangle
+                .construct(
+                    activation,
+                    &[
+                        scroll_rect.x_min.to_pixels().into(),
+                        scroll_rect.y_min.to_pixels().into(),
+                        (scroll_rect.x_max - scroll_rect.x_min).to_pixels().into(),
+                        (scroll_rect.y_max - scroll_rect.y_min).to_pixels().into(),
+                    ],
+                )?
+                .into());
+        } else {
+            return Ok(Value::Null);
+        }
+    }
+    Ok(Value::Undefined)
+}
+
+fn set_scroll_rect<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(dobj) = this.and_then(|this| this.as_display_object()) {
+        let rect = args[0].as_object().unwrap();
+        let x = rect
+            .get_property(&Multiname::public("x"), activation)?
+            .coerce_to_number(activation)?;
+
+        let y = rect
+            .get_property(&Multiname::public("y"), activation)?
+            .coerce_to_number(activation)?;
+
+        let width = rect
+            .get_property(&Multiname::public("width"), activation)?
+            .coerce_to_number(activation)?;
+
+        let height = rect
+            .get_property(&Multiname::public("height"), activation)?
+            .coerce_to_number(activation)?;
+
+        // Flash only updates the scrollRect when the next frame is rendered,
+        // resulting a brief window of time where 'DisplayObject.scrollRect'
+        // returns the newly-set rect, but methods like 'hitTestPoint',
+        // 'getBounds', and 'localToGlobal' will *not* take scrollRect into account.
+        // However, ActionScript code can't detect when a frame is rendered,
+        // so there "shouldn't" be any SWFs that actually rely on this behavior
+        // (since Flash might happen to render a frame immediately after scrollRect
+        // is updated). Therefore, we just have things like 'localToGlobal' update immediately
+        // (by reading from the 'scroll_rect' field) for simplicity.
+        //
+        // If it turns out that an SWF somehow depends on some non-deterministic number of
+        // 'localToGlobal' calls not taking scrollRect into account, then we'll need a more
+        // complicated implementation.
```

Comment:
Some questions here as I haven't looked into this behavior super thoroughly:

 * Is this simply a "one-frame" delay in most cases, where these functions will operate correctly the next frame, after the object renders?
 * There is [`Event.RENDER`](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display/DisplayObject.html#event:render), which is fired when the stage is about to be rendered (need to call `Stage.invalidate()` to trigger this). Is the `scrollRect` taken into account at that point?
 * Assuming this sets some "dirty" flag, are there any properties that force the object to be re-rendered? I would imagine things like `BitmapData.draw`, but possibility there are other properties, similar to text layout in #7749. I imagine we could use the `RenderContext::gc_context` added in that PR to properly update this during render eventually.

(In any case, I agree that we should ignore this behavior for now! Just clarifying for my own edification).

Reply:
I went ahead and implemented the 'update after render' behavior, since ActionScript can reliably observe the lack of updates in any code that runs after a `scrollRect` update (as long as it's not in an event handler / setTimeout).